tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -18,131 +18,124 @@
+                    Instance = new LiveDebugger(settings, discoveryService, lineProbeResolver, debuggerSink, probeStatusPoller, updater);
+                }
 
-            return debugger;
+                return Instance;
+            }
         }
 
-        private void Initialize()
+        public async Task InitializeAsync()
         {
-            if (!_settings.Enabled)
+            if (!ShouldInitialize())
             {
-                Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
                 return;
             }
 
-            if (_settings.TransportType != TracesTransportType.Default)
+            try
             {
-                Log.Information("Live Debugger is not supported on UDS or named pipelines.");
-                return;
-            }
+                Log.Information("Live Debugger initialization started");
 
-            Log.Information("Live Debugger initialization started");
+                Product.ConfigChanged += (sender, args) => AcceptConfiguration(args);
+                AppDomain.CurrentDomain.AssemblyLoad += (sender, args) => CheckUnboundProbes();
+                AppDomain.CurrentDomain.DomainUnload += (sender, args) => _lineProbeResolver.OnDomainUnloaded();
 
-            Task.Run(() => InitializeAsync());
+                await StartAsync().ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Initializing Live Debugger failed.");
+            }
 
-            AppDomain.CurrentDomain.AssemblyLoad += (sender, args) => CheckUnboundProbes();
-            AppDomain.CurrentDomain.DomainUnload += (sender, args) => _lineProbeResolver.OnDomainUnloaded();
+            bool ShouldInitialize()
+            {
+                if (!_settings.Enabled)
+                {
+                    Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
+                    return false;
+                }
 
-            Log.Information("Live Debugger initialization completed");
-        }
+                if (_settings.TransportType != TracesTransportType.Default)
+                {
+                    Log.Information("Live Debugger is not supported on UDS or named pipelines.");
+                    return false;
+                }
 
-        private async Task InitializeAsync()
-        {
-            try
-            {
-                var isDiscoverySuccessful = await _discoveryService.DiscoverAsync().ConfigureAwait(false);
-                var isProbeConfigurationSupported = isDiscoverySuccessful && !string.IsNullOrWhiteSpace(_discoveryService.ProbeConfigurationEndpoint);
-                if (_settings.ProbeMode == ProbeMode.Agent && !isProbeConfigurationSupported)
+                var isProbeConfigurationSupported = !string.IsNullOrWhiteSpace(_discoveryService.ConfigurationEndpoint);
```

Comment:
This variable name is a bit misleading - this endpoint is for RCM in general, not just probe configuration